Creating a selinux module to resolve selinux conflicts with grafana agent

falcocer-ibm opened the pull request:
This code creates a new ansible task in the grafana_agent role that configures a SELinux module on CentOS and RedHat systems which allows useradd command to access var_lib_t context files, that way we can avoid SELinux denials after the grafana installation on the teuthology-suite's executions.

Fixes: https://ibm.monday.com/boards/5591222586/pulses/8269716052

I have run a test with teuthology-suite command like this in order to verify if the module was working or not.
teuthology-suite -s smoke --ceph-repo ceph -c squid --suite-repo ceph -m smithi --subset 1/5 -p 50 -l 8 --force-priority --newest 10 --sha1 a8eeed19eb9ce9bdce3a02ea18f12005000c9146 /home/falcocer/ansible-branch.yml

Result before the module:
https://pulpito.ceph.com/falcocer-2025-02-17_21:50:23-smoke-squid-distro-default-smithi/

Result after the module:
https://pulpito.ceph.com/falcocer-2025-02-24_18:34:53-smoke-squid-distro-default-smithi/

falcocer-ibm commented on roles/grafana_agent/tasks/useradd-selinux.yml:
```diff
@@ -29,3 +29,12 @@
+
+- name: Debug SELinux module instalation
+  debug:
+    msg: "{{ semodule_list.stdout_lines }}"
```
The last teuthology-suite command was successful.
https://pulpito.ceph.com/falcocer-2025-02-24_18:34:53-smoke-squid-distro-default-smithi/

dmick commented on testnodes.yml:
```diff
@@ -4,4 +4,5 @@
   roles:
     - common
     - testnode
+    - grafana_agent
```
"Modifying"

dmick commented:
remember to rebase and squash the things where you're removing things that you added in these commits (the debug task, for example) before merging.  The history should contain only the final form with explanations of the new code.

falcocer-ibm commented:
I have squashed all my previous commits and updated my initial comment in order to provide a quick explanation of what this code does, please have a look @djgalloway and @dmick and let me know if there is anything else I should add before merging this into main.

dmick commented:
I'm concerned by the ansible lint failures but I don't understand them

djgalloway commented:
> I'm concerned by the ansible lint failures but I don't understand them

They're not related to the PR.  Something broken in the actual check config.  I'll fix.

zmc commented:
Let's please link issues that are publicly-readable in the future

djgalloway commented:
> Let's please link issues that are publicly-readable in the future

That's fair.  The context for this is I wanted to find where the system resource bottlenecks are on the smithi nodes when running teuthology jobs so I could plan for the new hardware purchase(s).
